aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -204,73 +185,51 @@ private AuthSession authSession() {
     String token = token().get();
     if (null != token) {
       return authSessionCache()
-          .get(
+          .cachedSession(
               token,
-              id ->
-                  AuthSession.fromAccessToken(
-                      httpClient(),
-                      tokenRefreshExecutor(),
-                      token,
-                      expiresAtMillis(properties()),
-                      new AuthSession(
-                          ImmutableMap.of(),
-                          AuthConfig.builder()
-                              .token(token)
-                              .credential(credential())
-                              .scope(SCOPE)
-                              .oauth2ServerUri(oauth2ServerUri())
-                              .optionalOAuthParams(optionalOAuthParams())
-                              .build())));
+              () -> {
+                Map<String, String> properties =
+                    ImmutableMap.<String, String>builder()
+                        .putAll(properties())
+                        .putAll(optionalOAuthParams())
+                        .put(OAuth2Properties.OAUTH2_SERVER_URI, oauth2ServerUri())
+                        .put(
+                            OAuth2Properties.TOKEN_REFRESH_ENABLED,
+                            String.valueOf(keepTokenRefreshed()))
+                        .put(OAuth2Properties.TOKEN, token)
+                        .put(OAuth2Properties.SCOPE, SCOPE)
+                        .buildKeepingLast();
+                return authManager().catalogSession(httpClient(), properties);
```

Comment:
should there maybe be an override of this method that takes `AuthConfig` instead of `properties`? And `catalogSession(..., Map<String, String> properties)` could then always call `catalogSession(..., AuthConfig)`

Reply:
I think not, because in spite of its generic name, `AuthConfig` is only meaningful for the OAuth2 scheme. It doesn't make sense e.g. for SigV4.